File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int leftPointer = 0;
        int rightPointer = numbers.size()-1;
        vector<int> result(2);
        while(leftPointer < rightPointer) {
            int sum = numbers[leftPointer] + numbers[rightPointer];
            if(sum < target) {
                leftPointer++;
            } else if(sum > target) {
                rightPointer--;
            } else {
                break;
            }
        }
        result[0] = leftPointer+1;
        result[1] = rightPointer+1;
        return result;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-2.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        # array of distinct integers: nums
        # target integer: target
        # repetition with replacement
        # combinations are same if frequency of each chosen numbers is same
        # return unique combinations of nums where the numbers sum up to target
        res = []
        subset = []
        def dfs(i, total):
            if total == target:
                res.append(subset.copy())
                return 
            if i >= len(nums) or total > target:
                return
            subset.append(nums[i])
            dfs(i, total + nums[i])
            subset.pop()
            dfs(i+1, total)
        dfs(0,0)
        return res